runtime/io.h

```diff
@@ -0,0 +1,163 @@
+    ExternalUnit, const char *format, std::size_t);
+Cookie IONAME(BeginExternalFormattedInput)(
+    ExternalUnit, const char *format, std::size_t);
+Cookie IONAME(BeginUnformattedOutput)(ExternalUnit);
+Cookie IONAME(BeginUnformattedInput)(ExternalUnit);
+Cookie IONAME(BeginNamelistOutput)(ExternalUnit, const NamelistGroup &);
+Cookie IONAME(BeginNamelistInput)(ExternalUnit, const NamelistGroup &);
+
+// Asynchronous I/O is supported (at most) for unformatted direct access
+// block transfers.
+AsynchronousID IONAME(BeginAsynchronousOutput)(
+    ExternalUnit, std::int64_t REC, const char *, std::size_t);
+AsynchronousID IONAME(BeginAsynchronousInput)(
+    ExternalUnit, std::int64_t REC, char *, std::size_t);
+Cookie IONAME(WaitForAsynchronousIO)(ExternalUnit, AsynchronousID);  // WAIT
+
+// Other I/O statements
+// TODO: OPEN & INQUIRE
+Cookie IONAME(BeginClose)(ExternalUnit);
+Cookie IONAME(BeginFlush)(ExternalUnit);
+Cookie IONAME(BeginBackspace)(ExternalUnit);
+Cookie IONAME(BeginEndfile)(ExternalUnit);
+Cookie IONAME(BeginRewind)(ExternalUnit);
+
+gggg  // Control list options
+    void IONAME(SetADVANCE)(Cookie, const char *, std::size_t);
+void IONAME(SetBLANK)(Cookie, const char *, std::size_t);
+void IONAME(SetDECIMAL)(Cookie, const char *, std::size_t);
+void IONAME(SetDELIM)(Cookie, const char *, std::size_t);
+void IONAME(SetPAD)(Cookie, const char *, std::size_t);
+void IONAME(SetPOS)(Cookie, std::int64_t);
+void IONAME(SetREC)(Cookie, std::int64_t);
+void IONAME(SetROUND)(Cookie, const char *, std::size_t);
+void IONAME(SetSIGN)(Cookie, const char *, std::size_t);
+
+// Data item transfer for modes other than namelist.
+// Any item can be transferred by means of a descriptor; unformatted
+// transfers to/from contiguous blocks can avoid the descriptor; and there
+// are specializations for the common scalar types.
+// Functions with Boolean results return false when the I/O statement
+// has encountered an error or end-of-file/record condition.
+bool IONAME(OutputDescriptor)(Cookie, const Descriptor &);
+bool IONAME(InputDescriptor)(Cookie, const Descriptor &);
+bool IONAME(OutputUnformattedBlock)(Cookie, const char *, std::size_t);
+bool IONAME(InputUnformattedBlock)(Cookie, char *, std::size_t);
+bool IONAME(OutputInteger64)(Cookie, std::int64_t);
+std::int64_t IONAME(InputInteger64)(Cookie, int kind = 8);
+bool IONAME(OutputReal32)(Cookie, float);
+float IONAME(InputReal32)(Cookie);
+bool IONAME(OutputReal64)(Cookie, double);
+double IONAME(InputReal64)(Cookie);
+bool IONAME(OutputComplex32)(Cookie, float, float);
+bool IONAME(OutputComplex64)(Cookie, double, double);
+bool IONAME(OutputASCII)(Cookie, const char *, std::size_t);
+bool IONAME(InputASCII)(Cookie, char *, std::size_t);
+
+// Result extraction after data transfers are complete.
+void IONAME(GetIOMSG)(Cookie, char *, std::size_t);  // IOMSG=
+void IONAME(GetSTATUS)(Cookie, char *, std::size_t);  // STATUS=
+int IONAME(GetIOSTAT)(Cookie);  // IOSTAT=
```

Comment:
An IOSTAT variable can be any kind.  This call should probably return int64_t.

Reply:
But the values that can be stored into `IOSTAT=` can all be represented as simple ints.  The compiler will take care of any necessary conversions.